docs/site/parinfer.md

```diff
@@ -1,18 +1,39 @@
 
-However, Calva's [Paredit](paredit.md) is carefully crafted. If you don't already have an established preference for parinfer, we suggest you give it a try. Even parinfer veterans may find they prefer the explicit nature of paredit. It takes only some few minutes to learn two or three basic Paredit commands:
+![Calva Parinfer](images/calva-parinfer-and-format-forward.gif)
 
-* **Select form**
-* **Slurp**
-* **Barf**
+## Infer Parens and Format Forward
+Calva has two experimental features: **Infer Parens** (as you type), and the corresponding **Format Forward** (as you type). For now, they are both disabled by default. Enable them via these settings:
 
-Also, if you find yourself having deleted or added a bracket out of structure, despite Calva Paredit's **Strict mode** (which you are using, right?) there is a command that might help you heal the structure: **Infer Parens from Indentation**. That command is actually implemented using the Parinfer library, so it will let you recover from quite many situations. This command is bound to `shift+tab` by default, making it a good companion to `tab` which indents based on bracket structure. 
+* `calva.fmt.experimental.inferParensAsYouType`
+* `calva.fmt.experimental.formatForward`
 
-All that said, what was said first is true: if you want to use the Parinfer extension, you can. You'll probably want to disable Calva's auto-formatting. The setting for controlling this is named `calva.fmt.formatAsYouType`.
+They go best together. The first letting you use indentation to decide structure, the latter using the structure to let you keep the code indented properly as you type.
+
+!!! Note "About `editor.autoClosingBrackets` and infering the closing bracket"
+    TL;DR: When enabling `calva.fmt.experimental.inferParensAsYouType`, _you need to also disable `editor.autoClosingBrackets`_.
+
+    For Parinfer's inference of the closing bracket when you type an opening one to work, VS Code's built-in `editor.autoClosingBrackets` setting needs to be **disabled**, and Calva currently **enables** this by default. This is because there are currently some troubles with Calva's backup auto-closing features (`calva.paredit.strictAutoClosingBrackets`).
+
+!!! Note "Multi-cursors not fully supported"
+    Calva only really considers the first cursor in a multi-cursor senario. Sometimes that's enough, often it is not.
+
+## Paredit is still there
+
+In Calva, Parinfer and [Paredit](paredit.md) are designed to coexist and both be there to let you edit the structure easily and efficiently. Since Pardit commands are always formatted, they leave the code in a state where Parinfer has what it needs to infer bracket placement as you either edit the indentation, or remove/add brackets.
```

Comment:
```suggestion
In Calva, Parinfer and [Paredit](paredit.md) are designed to coexist and both be there to let you edit the structure easily and efficiently. Since Paredit commands are always formatted, they leave the code in a state where Parinfer has what it needs to infer bracket placement as you either edit the indentation, or remove/add brackets.
```